src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -355,22 +356,34 @@
          (middleware-utils/json-response 200))))
 
 (defn describe-task-file
-  [env-name module-name file]
+  [get-code-content env-name code-id module-name file]
   (let [size (fs/size file)
         sha256 (ks/file->sha256 (io/file file))
-        base-name (fs/base-name file)]
+        base-name (fs/base-name file)
+        ;; we trust the file path from Puppet, so extract the subpath from file
+        static-path (re-find (re-pattern (str #"[^/]+/" module-name "/tasks/" base-name)) file)
+        uri (try
+              ;; if code content can be retrieved, then use static_file_content
+              (get-code-content env-name code-id static-path)
```

Comment:
If basemodulepath includes a directory that's not managed that contains the same module and task with different content, we'll get the wrong file (and pxp-agent will error downloading it because the SHA doesn't match).